monkey/monkey_island/cc/resources/propagation_credentials.py

```diff
@@ -35,7 +35,18 @@ def get(self, collection=None):
     @auth_token_required
     @roles_accepted(AccountRole.ISLAND_INTERFACE.name)
     def put(self, collection=None):
-        credentials = [Credentials(**c) for c in request.json]
+        try:
+            credentials = []
+            for credential_pair in request.json:
+                credentials.append(Credentials(**credential_pair))
+        except (TypeError, ValueError) as err:
+            return {
+                "error": {
+                    "message": f"{str(err)}",
+                    "bad_index": request.json.index(credential_pair),
+                }
+            }, HTTPStatus.BAD_REQUEST
```

Comment:
I'm assuming this is tested and `credential_pair` still exists and hasn't gone out of scope. If that's the case:

```suggestion
        try:
            credentials = []
            for index, credential_pair in enumerate(request.json):
                credentials.append(Credentials(**credential_pair))
        except (TypeError, ValueError) as err:
            return {
                "error": {
                    "message": f"{str(err)}",
                    "bad_index": index,
                }
            }, HTTPStatus.BAD_REQUEST
```
Otherwise:

```suggestion
        
        credentials = []
        for index, credential_pair in enumerate(request.json):
            try:
                credentials.append(Credentials(**credential_pair))
            except (TypeError, ValueError) as err:
                return {
                    "error": {
                        "message": f"{str(err)}",
                        "bad_index": index,
                    }
                }, HTTPStatus.BAD_REQUEST
```

Reply:
Yep, it still exists. We can go with the first one.